lib/block_view.go

```diff
@@ -3603,6 +3624,135 @@ func (bav *UtxoView) _connectTransaction(txn *MsgDeSoTxn, txHash *BlockHash,
+// failing transactions are included in the blocks and their fees are burned. Initially blocks will include full transactions,
+// and in the future iterations blocks will only include the failing transaction hashes. This is done to prevent spamming
+// attacks on the network with seemingly valid transactions that never commit and just end up occupying the block space.
+// In addition, a major part of the effective fees of this transaction is burned with BMF. This makes spam attacks economically
+// disadvantageous. Attacker's funds are burned, to the benefit of everyone else on the network. BMF algorithm also computes
+// a utility fee, which is distributed to the block producer.
+func (bav *UtxoView) _connectFailingTransaction(txn *MsgDeSoTxn, blockHeight uint32, verifySignatures bool) (
+	_utxoOps []*UtxoOperation, _burnFee uint64, _utilityFee uint64, _err error) {
+
+	// Failing transactions are only allowed after ProofOfStake2ConsensusCutoverBlockHeight.
+	if blockHeight <= bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight {
+		return nil, 0, 0, fmt.Errorf("_connectFailingTransaction: Failing transactions " +
+			"not allowed before ProofOfStake2ConsensusCutoverBlockHeight")
+	}
+
+	// Sanity check the transaction to make sure it is properly formatted.
+	if err := CheckTransactionSanity(txn, blockHeight, bav.Params); err != nil {
+		return nil, 0, 0, errors.Wrapf(err, "_connectFailingTransaction: "+
+			"Problem checking txn sanity")
+	}
+
+	if err := ValidateDeSoTxnSanityBalanceModel(txn, uint64(blockHeight), bav.Params, bav.GlobalParamsEntry); err != nil {
+		return nil, 0, 0, errors.Wrapf(err, "_connectFailingTransaction: "+
+			"Problem checking txn sanity under balance model")
+	}
+
+	if err := bav.ValidateTransactionNonce(txn, uint64(blockHeight)); err != nil {
+		return nil, 0, 0, errors.Wrapf(err, "_connectFailingTransaction: "+
+			"Problem validating transaction nonce")
+	}
+
+	// Get the FailingTransactionBMFRateBasisPoints from the global params entry. We then compute the effective fee
+	// as: effectiveFee = txn.TxnFeeNanos * FailingTransactionBMFRateBasisPoints / 10000
+	gp := bav.GetCurrentGlobalParamsEntry()
+
+	failingTransactionRate := uint256.NewInt().SetUint64(gp.FailingTransactionBMFRateBasisPoints)
+	failingTransactionFee := uint256.NewInt().SetUint64(txn.TxnFeeNanos)
+	basisPointsAsUint256 := uint256.NewInt().SetUint64(10000)
+
+	effectiveFeeU256 := failingTransactionRate.Mul(failingTransactionRate, failingTransactionFee)
+	effectiveFeeU256.Div(effectiveFeeU256, basisPointsAsUint256)
+	// We should never overflow on the effective fee, since FailingTransactionBMFRateBasisPoints is <= 10000.
+	// But if for some magical reason we do, we set the effective fee to the max uint64. We don't error, and
+	// instead let _spendBalance handle the overflow.
+	maxUint64 := uint256.NewInt().SetUint64(math.MaxUint64)
+	if effectiveFeeU256.Cmp(maxUint64) > 0 {
+		effectiveFeeU256.SetUint64(math.MaxUint64)
+	}
```

Comment:
The uint256 package has a nice little IsUInt64() function for this use
```suggestion
	if !effectiveFeeU256.IsUint64() {
		effectiveFeeU256 = uint256.NewInt().SetUint64(math.MaxUint64)
	}
```